pom.client.xml

```diff
@@ -547,17 +535,74 @@
 
    <profiles>
 
+       <!--
+        By default we build against our baseline, Java 8, but we also want to ensure compatibility
+        against the latest Java LTS release, currently Java 11. We therefore have two profiles:
+          * The default 'java8', which will perform a build using Java 8 as its target.
+          * The non-default 'java-lts' profile, which will build against the current LTS release.
+      -->
       <profile>
+         <id>java8</id>
+         <activation>
+            <activeByDefault>true</activeByDefault>
+         </activation>
+         <build>
+            <plugins>
+               <plugin>
+                  <groupId>org.apache.maven.plugins</groupId>
+                  <artifactId>maven-compiler-plugin</artifactId>
+                  <version>${maven-compiler-plugin.version}</version>
+                  <configuration>
+                     <source>1.8</source>
+                     <target>1.8</target>
+                     <showWarnings>true</showWarnings>
+                     <showDeprecation>true</showDeprecation>
+                     <compilerArgument>-Xlint:all</compilerArgument>
+                  </configuration>
+               </plugin>
+            </plugins>
+         </build>
+      </profile>
 
+      <profile>
+         <id>java-lts</id>
+         <build>
+            <plugins>
+               <plugin>
+                  <groupId>org.apache.maven.plugins</groupId>
+                  <artifactId>maven-compiler-plugin</artifactId>
+                  <version>${maven-compiler-plugin.version}</version>
+                  <configuration>
+                     <source>11</source>
+                     <target>11</target>
+                     <showWarnings>true</showWarnings>
+                     <showDeprecation>true</showDeprecation>
+                     <compilerArgument>-Xlint:all</compilerArgument>
+                  </configuration>
+               </plugin>
+
+               <!-- Customise the javadoc plugin to not include module directories in links. We don't
+                    currently use modules, so search results were being directed to an 'undefined'
+                    directory (for an 'undefined' module). This --no-module-directories flag fixes 
```

Comment:
The `--no-module-directories` bit here is breaking the build. I guess two dashes can only be used inside of a comment to indicate the end of the comment block...

Reply:
Ha. I'll fix - thanks